rollout/controller.go

```diff
@@ -251,9 +259,32 @@ func NewController(cfg ControllerConfig) *Controller {
 			controllerutil.EnqueueParentObject(obj, register.RolloutKind, controller.enqueueRollout)
 		},
 	})
+
+	cfg.IstioVirtualServiceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			controller.processNextEnqueuedObj(obj)
+		},
+		// TODO: DeepEquals on httpRoutes
+		UpdateFunc: func(old, new interface{}) {
+			controller.processNextEnqueuedObj(new)
+		},
+		DeleteFunc: func(obj interface{}) {
+			controller.processNextEnqueuedObj(obj)
+		},
+	})
+
 	return controller
 }
 
+func (c *Controller) DoesIstioExist() bool {
+	gvk := schema.ParseGroupResource("virtualservices.networking.istio.io").WithVersion(c.defaultIstioVersion)
+	_, err := c.dynamicclientset.Resource(gvk).List(metav1.ListOptions{})
```

Comment:
This list should respect the `--namespace` flag to the controller. So I think this should be:

```golang
_, err := c.dynamicclientset.Resource(gvk).Namespace(c.namespace).List(metav1.ListOptions{})

```